.github/workflows/twister_dfu_nrf52840dk.yml

```diff
@@ -0,0 +1,138 @@
+name: Twister DFU nrf52840dk
+
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    # Run workflow at the start of every day (12 AM UTC)
+    - cron: "0 0 * * *"
+
+  workflow_dispatch:
+
+jobs:
+  # Assumptions made about the self-hosted runner:
+  #
+  #  1. Has installed the GitHub Actions self-hosted runner service
+  #  2. Has nrfjprog already installed at /opt/nrf-command-line-tools
+  #  3. Has SEGGER tools already installed at /home/golioth/JLink_Linux_V786_arm64
+  #  4. GitHub secret YOUR_RUNNER_NAME_NRF52840DK_PORT
+  #  5. GitHub secret YOUR_RUNNER_NAME_NRF52840DK
+  #  6. GitHub secret GOLIOTHCTL_YAML
+  #
+  # It is the responsibility of the self-hosted runner admin to ensure
+  # these pre-conditions are met.
+  #
+  # For item 1, GitHub provides instructions when you add a new self-hosted runner
+  # in Settings -> Actions -> Runners.
+  #
+  # For item 2, you'll need to install jlink and nrf-command-line-tools:
+  #   https://www.segger.com/downloads/jlink/
+  #   https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools/download
+  #
+  # For item 3, this environment variable can be added to $HOME/runner_env.sh:
+  #   export CI_NRF52840DK_PORT=/dev/ttyACM0
+  #
+  # For item 4, the file needs to have contents like (use base64 encoding):
+  #
+  # - connected: true
+  #   id: "001050257254"
+  #   platform: nrf52840dk_nrf52840
+  #   product: J-Link
+  #   runner: nrfjprog
+  #   serial: /dev/serial/by-id/usb-SEGGER_J-Link_001050257254-if00
+  #
+  # For item 5, the file needs to have contents like (use base64 encoding):
+  #
+  # export GOLIOTH_SYSTEM_SERVER_HOST=coap.golioth.dev
+  # export GOLIOTH_SYSTEM_CLIENT_PSK_ID=device_psk_id@ci
+  # export GOLIOTH_SYSTEM_CLIENT_PSK=device_psk
+  # export GOLIOTH_SAMPLE_WIFI_SSID=golioth-runner-xxx
+  # export GOLIOTH_SAMPLE_WIFI_PSK=password_for_wifi
+  # export GOLIOTH_DEVICE_NAME=nrf52840dk
+  # export GOLIOTHCTL_CONFIG=$HOME/.goliothctl-local.yaml
```

Comment:
This is not needed if the file will be located at `$HOME/.golioth/.goliothctl.yaml` (this is the default path).